newsroom/products/products.py

```diff
@@ -55,6 +58,28 @@ def on_deleted(self, doc: Product) -> None:
                 updates = {"products": [p for p in item["products"] if p["_id"] != doc["_id"]]}
                 superdesk.get_resource_service(resource).system_update(item["_id"], updates, item)
 
+    def create(self, docs):
+        company_products = {}
+        for doc in docs:
+            if doc.get("companies"):
+                for company_id in doc["companies"]:
+                    company_products.setdefault(company_id, []).append(doc)
+        res = super().create(docs)
+        if company_products:
+            warnings.warn("Using deprecated product.companies", DeprecationWarning)
```

Comment:
This was made obsolete in v2.4. Do we still have code using this somewhere?
IMO, it would make for cleaner/stricter code if we removed support for this

Reply:
it got quite complicated to fix all tests, especially those behave ones, otherwise this shouldn't be used, but will need to make some changes in superdesk core steps to make it simpler and that will go into 2.7